session.go

```diff
@@ -727,33 +720,36 @@ func (s *Session) routingKeyInfo(ctx context.Context, stmt string) (*routingKeyI
 	return routingKeyInfo, nil
 }
 
-func (b *Batch) execute(ctx context.Context, conn *Conn) *Iter {
-	return conn.executeBatch(ctx, b)
-}
-
 // Exec executes a batch operation and returns nil if successful
 // otherwise an error is returned describing the failure.
 func (b *Batch) Exec() error {
 	iter := b.session.executeBatch(b)
 	return iter.Close()
 }
 
+// Iter executes a batch operation and returns an Iter object
+// that can be used to access properties related to the execution like Iter.Attempts and Iter.Latency
+func (b *Batch) Iter() *Iter {
+	return b.session.executeBatch(b)
```

Comment:
I feel that people may execute sequence of `Exec()` and `Iter()`. Shall we make an assertion that only one of the is invoked?

Reply:
> I feel that people may execute sequence of `Exec()` and `Iter()`. Shall we make an assertion that only one of the is invoked?

They can't execute sequence (`Exec().Iter()`) unless they do it like:
```
err := b.Exec()
iter := b.Iter()
``` 

Which I believe is already something that can be done with the current API and it will trigger two requests. We can try to improve documentation if this is a concern.

+1 to what James said